apisix/core/resolver.lua

```diff
@@ -19,15 +19,67 @@
 --
 -- @module core.resolver
 
-local json = require("apisix.core.json")
-local log = require("apisix.core.log")
-local utils = require("apisix.core.utils")
+local json           = require("apisix.core.json")
+local log            = require("apisix.core.log")
+local utils          = require("apisix.core.utils")
+local string_match   = string.match
+local open           = io.open
+
+
+local DEFAULT_HOSTS = "/etc/hosts"
+local HOSTS_IP_MATCH_CACHE = {}
 
 
 local _M = {}
 
 
+local function guess_name_type(name)
+    if name:match("^[%d%.]+$") then
+        return "ipv4"
+    end
+
+    if name:find(":") then
+        return "ipv6"
+    end
+
+    return "hostname"
+end
+
+
+local function init_hosts_ip()
```

Comment:
Could we use the method from https://github.com/Kong/lua-resty-dns-client/blob/fb1e686a8857c7492fdd30ca228d200afbde19aa/src/resty/dns/client.lua#L493

Reply:
Yeah,  I will replace it